src/cli.ts

```diff
@@ -182,3 +191,60 @@ function prepareCliMessage(
 
 	return spaced ? `\n${preparedMessage}\n` : preparedMessage;
 }
+
+export async function printEnvInfo(): Promise<void> {
+	// eslint-disable-next-line no-console
+	console.log(
+		`
+System:
+  Platform: ${os.platform()}
+  Arch: ${os.arch()}
+  Version: ${os.version()}
+  CPU: (${os.cpus().length}) ${os.arch()} ${os.cpus()[0]?.model}
+  Memory: ${Math.round(os.totalmem() / 1024 / 1024 / 1024)} GB
+  Shell: ${process.env['SHELL']?.toString() ?? 'Unknown'}
+Binaries:
+  Node: ${process.versions.node}
+  Yarn: ${getBinaryVersion('yarn')}
+  npm: ${getBinaryVersion('npm')}
+  pnpm: ${getBinaryVersion('pnpm')}
+Package Manager Used: ${await getCurrentPackageManager()}
+Relevant Packages:
+  @cloudflare/next-on-pages: ${nextOnPagesVersion}
+  vercel: ${getPackageVersion('vercel')}
+  next: ${getPackageVersion('next')}
+  react: ${getPackageVersion('react')}
+  react-dom: ${getPackageVersion('react-dom')}
+  eslint-config-next: ${getPackageVersion('eslint-config-next')}
+`.trim()
+	);
+}
```

Comment:
I would suggest the following, so that:
 - we get extra lines before and after the info, which does look nice I think?
 - no eslint-config-next info, I'm not totally sure we care about it (we could probably also remove react and react-dom 🤷)
 - we can indent the message in the code however we like

```suggestion
export async function printEnvInfo(): Promise<void> {
	const envInfoMessage = dedent(`
		System:
			Platform: ${os.platform()}
			Arch: ${os.arch()}
			Version: ${os.version()}
			CPU: (${os.cpus().length}) ${os.arch()} ${os.cpus()[0]?.model}
			Memory: ${Math.round(os.totalmem() / 1024 / 1024 / 1024)} GB
			Shell: ${process.env['SHELL']?.toString() ?? 'Unknown'}
		Binaries:
			Node: ${process.versions.node}
			Yarn: ${getBinaryVersion('yarn')}
			npm: ${getBinaryVersion('npm')}
			pnpm: ${getBinaryVersion('pnpm')}
		Package Manager Used: ${await getCurrentPackageManager()}
		Relevant Packages:
			@cloudflare/next-on-pages: ${nextOnPagesVersion}
			vercel: ${getPackageVersion('vercel')}
			next: ${getPackageVersion('next')}
			react: ${getPackageVersion('react')}
			react-dom: ${getPackageVersion('react-dom')}
	`);

	// eslint-disable-next-line no-console
	console.log(`\n${envInfoMessage}\n`);
}
```

@tarunrajput what do you think? 🙂

Reply:
Sure, I agree with your suggestions. I have updated the PR with these changes.

Reply:
Awesome, thanks! 😄